exla/livebooks/rotating_image.livemd

```diff
@@ -176,31 +178,45 @@ defmodule Rotate do
     {n, 2} = Nx.shape(pos)
 
     preprocessed_pos = preprocess_position(pos, n)
-    image = image |> Nx.reshape({n * 4})
+    image = image |> Nx.reshape({n * @num_channels})
 
     Nx.indexed_add(new_image, preprocessed_pos, image)
   end
 end
 ```
 
-## Example of usage
+## Example of usage with Lena image
 
 ```elixir
+@num_channels 3
+
 {_, dir} = File.cwd()
-path_to_file = "#{dir}/<image>.png"
-path_to_save = "#{dir}/<image_rotated>.png"
 
-image = Preprocess.image_to_tensor(path_to_file)
+path_to_file =
+  "https://www.researchgate.net/profile/Adelrahman-Idries/publication/232815017/figure/fig3/AS:300463506378754@1448647490218/Original-Lena-image.png"
```

Comment:
```suggestion
  "https://www.researchgate.net/publication/259521525/figure/fig8/AS:268024322195479@1440913386352/Original-standard-test-image-of-Peppers.png"
```

Reply:
This could be a `Kino.Input.url("Image to rotate")` and we can use the URL above as default. You can use `Kino.Input.read` to read the configured URL. :)

Reply:
I'm thinking right now about how we can combine both URLs and paths to files. The version below works cool for URLs, but is there any way to combine this with paths? Or is it not a case and just leave it?
```elixir
default_url =
   "https://www.researchgate.net/publication/259521525/figure/fig8/AS:268024322195479@1440913386352/Original-standard-test-image-of-Peppers.png"


input = Kino.Input.url("Image to rotate", default: default_url)
Kino.render(input)
url = Kino.Input.read(input)
```

Btw is it possible to expand the box generated by `Kino.input.read` a little bit?

Reply:
Ah, since we want both file and url we should just use Kino.Input.text instead.

it is not possible to expand the text size right now but we can add this in the future!

Reply:
Code wise, you can use URI.parse, I posted a suggestion elsewhere!